proto/spawn.proto

```diff
@@ -252,33 +262,106 @@ message ExecLogEntry {
   }
 
   // A set of spawn inputs.
-  // The contents of the set are the directly referenced files, directories and
-  // symlinks in addition to the contents of all transitively referenced sets.
+  // The contents of the set are the directly contained entries in addition to
+  // the contents of all transitively referenced sets. When order matters,
+  // transitive sets come before direct entries and within a set, entries are
+  // considered in left-to-right order ("postorder").
   // Sets are not canonical: two sets with different structure may yield the
   // same contents.
   message InputSet {
-    // Entry IDs of files belonging to this set.
-    repeated int32 file_ids = 1;
-    // Entry IDs of directories belonging to this set.
-    repeated int32 directory_ids = 2;
-    // Entry IDs of unresolved symlinks belonging to this set.
-    repeated int32 unresolved_symlink_ids = 3;
-    // Entry IDs of other sets contained in this set.
-    repeated int32 transitive_set_ids = 4;
+    // Entry IDs of files, directories, unresolved symlinks or runfiles trees
+    // belonging to this set.
+    repeated uint32 input_ids = 5;
+    // Entry IDs of other input sets contained in this set.
+    repeated uint32 transitive_set_ids = 4;
+
+    // Merged into input_ids for backwards compatibility with Bazel < 7.4.0.
+    repeated uint32 file_ids = 1 [deprecated = true];
+    repeated uint32 directory_ids = 2 [deprecated = true];
+    repeated uint32 unresolved_symlink_ids = 3 [deprecated = true];
+  }
```

Comment:
We might need to also modify the cli/print package to use the new proto fields? Perhaps in a separate PR though.

Reply:
Yeah, but that's going to be a pretty large change by itself as we also need to flatten the new `RunfilesTree` message. I will build it into `explain` first as the expanded execution log will probably be deprecated in Bazel 8, but we could still look into making `printlog` work for both.